src/katsdpcontroller/product_controller.py

```diff
@@ -862,17 +862,24 @@ def observer(sensor, reading):
             return
 
         logger.info(f"Waiting for {len(sensors)} tasks to receive good data")
```

Comment:
My suggestion is that if the user has asked not to have the timeout, then we shouldn't even try to wait i.e. put the rest of the function from this point inside an `if` statement. If someone is using this option, then there is a high chance it's going to fail, and we might as well not make them wait an extra 30 seconds in their debug cycle.

Reply:
I guess the only disadvantage to that is that you don't get the log message but as you point out, if you're using this option, then I guess you already know something like this is coming and it's not necessary. Fair point.